plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts

```diff
@@ -229,25 +229,27 @@ type StockStatusCount = {
 	count: number;
 };
 
+type PriceRangeProps = {
+	min_price: string;
+	max_price: string;
+	currency_code: CurrencyCode;
+	currency_decimal_separator: '.' | string;
+	currency_minor_unit: number;
+	currency_prefix: '$' | string;
+	currency_suffix: '' | string;
+	currency_symbol: '$' | string;
+	currency_thousand_separator: ',' | string;
+};
```

Comment:
Let's do it 👍

Reply:
what about predefined types

For instance:

```ts
currency_prefix: '$' | string;
```

It defines the `$`? Could we move that in the `CurrencyResponse` type def?

Reply:
Can you explain why do we need a default value for the prefix?

Reply:
And other properties, too. It's not clear to me why we need them.

Reply:
> It doesn't mean that is not possible to use a regular string, tho.

From your screenshot, it gives me the impression that the property is a union type with 5 available values.

Reply:
But isn't it against the original purpose — offering convenience and clarity? 

For example, from the suggestion, I think it's a union type. But then I figured out: Oh, it's not; it's a string. It seems to me that we're losing clarity here. Is there any issue with my thought process? Is my assumption on the union type correct? Happy to change my mind if I interpret it wrong.

Reply:
Also, maybe we should loop more people in for more diverse opinions?

Reply:
I'm not convinced about listing the currency symbols for two reasons:

-  list is incomplete (found at least 2 missing currencies supported by WordPress: `New Taiwan Dollar NT$ - NTD` or `Yemeni rial ﷼ - YER` but there's more).
- list doesn't support three letter codes (e.g. EUR, PLN etc.) Some currencies don't have symbols at all and only use code ISO 4217 codes.

So to sum up I'm either up for full list (which I don't think we really need, it doesn't bring much value compared to effort) or keeping it `string` as it was 😅

Reply:
> But isn't it against the original purpose — offering convenience and clarity?

I don't think so. It keeps offering convenience and clarity and now also suggests valid typical values.

> For example, from the suggestion, I think it's a union type.

It is!

> But then I figured out: Oh, it's not; it's a string.

It's a union type: `literals[] ` + `string`.

> It seems to me that we're losing clarity here. Is there any issue with my thought process? Is my assumption on the union type correct? Happy to change my mind if I interpret it wrong.

To me, it's pretty clear and straightforward. The type has valid common types but also has the flexibility to define any other values if needed.

Reply:
> Also, maybe we should loop more people in for more diverse opinions?

Yes, or maybe I should roll back these changes to move on.

Reply:
> I'm not convinced about listing the currency symbols for two reasons:
> 
> * list is incomplete (found at least 2 missing currencies supported by WordPress: `New Taiwan Dollar NT$ - NTD` or `Yemeni rial ﷼ - YER` but there's more).

The list was composed by picking the symbols from here [woocommerce/i18n/locale-info.php](https://github.com/woocommerce/woocommerce/blob/e62c541d8f3adb8568bc33fd30c729185dbdff7f/plugins/woocommerce/i18n/locale-info.php). 

Also, that's the point of this approach: although the list is not complete, something that we can work on is adding other code, symbols, etc., it's possible to use any `string` value.

> * list doesn't support three letter codes (e.g. EUR, PLN etc.) Some currencies don't have symbols at all and only use code ISO 4217 codes.

It's just a list of literal values. You can use whatever you need since, in the end, it also accepts strings.
 
> So to sum up I'm either up for full list (which I don't think we really need, it doesn't bring much value compared to effort) or keeping it `string` as it was 😅

There is no need to keep the list updated. All currency properties also accept random generica values.

Reply:
I can see the benefit of having the editor suggest values, but I'm still not convinced. In VSCode, I get this when trying to check the type of `currency_symbol` property:

<img width="1707" alt="image" src="https://github.com/user-attachments/assets/c92a18e6-981a-427c-b6fe-96892a880714">

I can't figure out the exact type of `currency_symbol` property by hovering on it. I can't click on the more button in the middle to expand the list, too.

By the way, does it have `undefined` type?

I tried reducing the suggestion list to 6 values, and here is what I got

<img width="1472" alt="image" src="https://github.com/user-attachments/assets/76d89001-cad6-410b-b7e4-dc3b42cc553a">

To me, it's a very complex type that I can't understand without going to the type definition and some effort (to understand `string & {}` type).

In summary, the benefit of the complex type here doesn't outperform the complexity it adds to me. So I'm not convinced.